src/account/account-data.ts

```diff
@@ -0,0 +1,125 @@
+import { Bee, BeeDebug } from '@ethersphere/bee-js'
+import { Wallet } from 'ethers'
+import { assertAddress, assertMnemonic, assertPassword, assertUsername } from './utils'
+import { prepareEthAddress } from '../utils/address'
+import { getEncryptedMnemonic } from './mnemonic'
+import { decrypt } from './encryption'
+import { createUser, UserAccountWithReference } from './account'
+
+export class AccountData {
+  /** username -> ethereum wallet address mapping */
+  public readonly usernameToAddress: { [key: string]: string } = {}
+
+  constructor(public readonly bee: Bee, public readonly beeDebug: BeeDebug, public wallet?: Wallet) {}
+
+  /**
+   * Sets the current account's wallet to interact with the data
+   *
+   * @param wallet BIP-039 + BIP-044 Wallet
+   */
+  setActiveAccount(wallet: Wallet): void {
+    this.wallet = wallet
+  }
+
+  /**
+   * Import FDP user account
+   *
+   * @param username Username to import
+   * @param mnemonic 12 space separated words to initialize wallet
+   */
+  async import(username: string, mnemonic: string): Promise<void> {
+    assertUsername(username)
+    try {
+      assertMnemonic(mnemonic)
+      const wallet = Wallet.fromMnemonic(mnemonic)
+      this.usernameToAddress[username] = prepareEthAddress(wallet.address)
+      this.setActiveAccount(wallet)
+    } catch (e) {
+      throw new Error('Incorrect mnemonic')
+    }
+  }
+
+  /**
+   * Set Ethereum address for specific username
+   *
+   * @param username Username to modify
+   * @param address Ethereum address with or without 0x prefix
+   */
+  setUserAddress(username: string, address: string): void {
+    assertUsername(username)
+    address = prepareEthAddress(address)
+    assertAddress(address)
+    this.usernameToAddress[username] = address
+  }
+
+  /**
+   * Remove Ethereum address for specific username
+   *
+   * @param username Username to modify
+   */
+  removeUserAddress(username: string): void {
+    assertUsername(username)
+    this.usernameToAddress[username] = ''
+  }
+
+  /**
+   * Logs in with the FDP credentials and gives back ethers wallet
+   *
+   * @param username FDP username
+   * @param password password of the wallet
+   * @returns BIP-039 + BIP-044 Wallet
+   */
+  async login(username: string, password: string): Promise<Wallet> {
```

Comment:
I would add an optional parameter for `address` that falls back to `this.usernameToAddress` in order to login in one line